File: itinerary_server/src/main/java/com/dametto/itinerary_server/jpa/Comment.java
package com.dametto.itinerary_server.jpa;

import java.util.Date;

public class Comment {
    private Integer id;

    private ShallowUser author;

    private String body;

    private Date createdAt;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public ShallowUser getAuthor() {
        return author;
    }

    public void setAuthor(ShallowUser author) {
        this.author = author;
    }

    public String getBody() {
        return body;
    }

    public void setBody(String body) {
        this.body = body;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }
}

File: itinerary-fe/src/app/models/comment.ts
import * as moment from "moment";
import { User } from "./user";

export class Comment {
    public id: number;
    public author: User;
    public body: string;
    public createdAt: moment.Moment;

    constructor(id: number, author: User, body: string, createdAt: moment.Moment) {
        this.id = id;
        this.author = author;
        this.body = body;
        this.createdAt = createdAt;
    }
}